src/zimcheck.cpp

```diff
@@ -109,6 +109,26 @@ class ErrorLogger {
     }
 };
 
+// There is exactly the same code in zimwriterfs
+inline std::string removeLocalTagAndParameters(const std::string& url)
+{
+  std::string retVal = url;
+  std::size_t found;
+
+  /* Remove URL arguments */
+  found = retVal.find("?");
+  if (found != std::string::npos) {
+    retVal = retVal.substr(0, found);
+  }
+
+  /* Remove local tag */
+  found = retVal.find("#");
+  if (found != std::string::npos) {
+    retVal = retVal.substr(0, found);
+  }
```

Comment:
This two search can be merged with a call to `std::string::find_first_of` (https://en.cppreference.com/w/cpp/string/basic_string/find_first_of)

`retVal.find_first_of("?#")`

And as we don't modify url, we don't need to copy url in retVal. In fact, we don't need a retVal at all :

```
auto found = url.find_firt_of("?#");
if (found != std::string::npos) {
    return url.substr(0, found);
}
return url;
```